### DISCUSSION TITLE
HTTP Authentication

### DISCUSSION THREAD

**Discussion by aoturoa** [rvdbreemen/OTGW-firmware#214] [Ideas]
Hi,

A huge thank you to Robert for all the hard work and contributions.

Would it be an idea to add a simple HTTP authentication to the Web Interface?
I have the OTGW connected to a WiFi network. The OTGW hostname and ip address are available to anyone on the network. 

The firmware could provide a default username / password . Those values could then be adjusted via Settings.

Looking forward to your thoughts on this subject.

**Comment by 0crap** [comment]
Very good idea!
We all know HTTPS is the way to go, but as a first step up I like the idea of simple HTTP authentication versus nothing at all.

**Reply by rvdbreemen** [reply]
HTTPS in a home use scenario is overkill. WiFi is already encrypted. I will not spend time on https myself. 

Then to the authentication question. I agree that it would be a benefit. I however think all IOT devices should also be on a separate VLAN or SSID and not exposed on your normal WiFi. 

I am not investing my time in this feature. But I will accept and consider a pull request of someone is willing to spend their time on this.

**Reply by aoturoa** [reply]
I agree that HTTPS is overkill.
The idea is to add basic authentication to prevent regular users on the network snooping around and adjust settings.
I am not talking about hackers who want you to burn more gas or put you in the cold.

LAN separation for IOT is a good practice.

**Reply by rvdbreemen** [reply]
True that it would benefit from a little authentication. Personally I am not worried about this scenario at this point, adjusting settings is a risk agreed. It could break a functioning system. 

If anyone has the bandwidth aka time to spend. I will accept a pull request.
